### PR TITLE
MS Graph Mail - added reply-mail command

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33997

## Description
Added the generic command reply-mail to replies to an email using Graph Mail.

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 1]
```diff
@@ -1464,6 +1481,51 @@ def send_email_command(client: MsGraphClient, args):
...
+
+def reply_email_command(client: MsGraphClient, args):
+    """
+    Reply to an email from user's mailbox, the sent message will appear in Sent Items folder
+    """
+    email_to = argToList(args.get('to'))
+    email_from = args.get('from', client._mailbox_to_fetch)
+    message_id = args.get('inReplyTo')
+    email_body = args.get('body', "")
+    email_subject = 'Re: ' + args.get('subject', "")
```
Comment (guykeller) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:1512]: wouldn't using fstring be better?

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 2]
```diff
@@ -1464,6 +1481,51 @@ def send_email_command(client: MsGraphClient, args):
...
+    attach_ids = argToList(args.get('attachIDs'))
+    email_cc = argToList(args.get('cc'))
+    email_bcc = argToList(args.get('bcc'))
+    html_body = args.get('htmlBody')
+    attach_names = argToList(args.get('attachNames'))
+    attach_cids = argToList(args.get('attachCIDs'))
+    message_body = html_body if html_body else email_body
```
Comment (guykeller) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:1519]:
```suggestion
    message_body = html_body or email_body
```

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 3]
```diff
@@ -1464,6 +1481,51 @@ def send_email_command(client: MsGraphClient, args):
...
+
+    suffix_endpoint = f'/users/{email_from}/messages/{message_id}/reply'
+    reply = client.build_message_to_reply(email_to, email_cc, email_bcc, email_subject, message_body, attach_ids,
+                                          attach_names, attach_cids)
+    client.ms_client.http_request('POST', suffix_endpoint, json_data={'message': reply}, resp_type="text")
```
Comment (guykeller) [inline Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:1524]: What is the response here? Is there a possibility of a failed request with a non-failure error code?

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py (hunk 3 above)
Comment (abaumgarten) [reply Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py:1525]: There is no useful response here, error handling of that request is done in MicrosoftApiModule (MicrosoftClient.http_request).

**Comment by abaumgarten** [comment]
@guykeller I had to add classifier and a mapper

**Review by guykeller — APPROVED** [review]

**Pull request merged**